.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '29 5 * * 4'
```

Comment:
Just out of curiousity; why do we execute this code scanning on a schedule basis as well ?  If you do it on every completed PR on master, I don't see any necessity do to this on a schedule basis as well ?

Reply:
Hmm, yeah, that is correct. Hmm, maybe this is bc the CodeQL scanning database is getting updated outside PR's. So, if a new security vulnerability surfaces, we don't have to wait for someone to make a PR to receive the notification.

Reply:
Good point.